### PR TITLE
Toggle task compensation and loop markers

### LINKED ISSUE #1271: Add radio toggles for task markers in admin accordion
Tasks:
- [x] Create a custom inspector control in `src/components/inspectors`.
- [x] Add the inspector under the Advanced accordion
- [x] Add a radio group to the control with the following options:
  - Parallel multi-instance
  - Sequential multi-instance
  - Loop
  - No Loop
- [x] Add checkbox for `isForCompensation` property
- [x] Add custom `inspectorHandler` to `task/index.js` config to handle mapping of radio value to BPMN definition values.
- [x] Add custom `inspectorData` to `task/index.js` config to handle mapping of BPMN definition values to radio group value.

### PR DESCRIPTION
**Pull request by vehikl-jacare**
Fixes #1271 

This PR adds radio buttons for Task Loop Characteristics in the advanced inspector:
Screen Shot 2020-12-01 at 10 43 40 AM

This produces the following bpmn: 
``` xml
<bpmn:task id="node_1" name="Form Task" pm:assignment="requester">
      <bpmn:standardLoopCharacteristics />
</bpmn:task>
```

It also adds a "For Compensation" checkbox to set the `isForCompensation` property on the task:
Screen Shot 2020-12-01 at 10 42 50 AM

This produces the following bpmn:
```xml
<bpmn:task id="node_1" name="Form Task" isForCompensation="true" pm:assignment="requester" />
```

### CODE REVIEW

File: src/components/nodes/task/index.js [hunk 1]
```diff
@@ -29,6 +32,56 @@ export default {
...
+          setNodeProp(node, 'loopCharacteristics', moddle.create('bpmn:MultiInstanceLoopCharacteristics'));
+        }
+
+        if (value.markerFlags.loopCharacteristics === 'sequential_mi') {
+          if (currentLoopCharacteristics.$type === 'bpmn:MultiInstanceLoopCharacteristics' && currentLoopCharacteristics.isSequential){
+            return;
+          }
+          setNodeProp(node, 'loopCharacteristics', moddle.create('bpmn:MultiInstanceLoopCharacteristics', {isSequential: true}));
+        }
+      }
```
Comment (vehikl-jacare) [inline src/components/nodes/task/index.js:61]: Fairly deep in the conditionals here

File: src/components/nodes/task/index.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import component from './task.vue';
 import nameConfigSettings from '@/components/inspectors/nameConfigSettings';
-import { taskHeight, taskWidth } from './taskConfig';
-import advancedAccordionConfig from '@/components/inspectors/advancedAccordionConfig';
+import {taskHeight, taskWidth} from './taskConfig';
```
Comment (vehikl-jacare) [inline src/components/nodes/task/index.js:3]: spacing

File: src/components/nodes/task/index.js [hunk 3]
```diff
@@ -49,8 +102,28 @@ export default {
...
+  if (loopCharacteristics.$type === 'bpmn:MultiInstanceLoopCharacteristics' && !loopCharacteristics.isSequential) {
+    return 'parallel_mi';
+  }
+
+  if (loopCharacteristics.$type === 'bpmn:MultiInstanceLoopCharacteristics' && loopCharacteristics.isSequential) {
+    return 'sequential_mi';
+  }
+
+  return 'no_loop';
+}
```
Comment (vehikl-jacare) [inline src/components/nodes/task/index.js:129]: This code is very similar to the code in the `inspectorHandler` method above. We could possibly pull it out into a function or some type of map object that associates `$type` to the inspector panel radio input values.

**Comment by caleeli** [comment]
I've do quick test. I found an issue when setting the task properties after changing the loop characteristics.

InspectorTask mp4

**Comment by caleeli** [comment]
Review inspector handler after loading:
ReviewInspectorAfterLoading mp4

File: src/components/nodes/task/index.js (hunk 3 above)
Comment (vehikl-jacare) [reply src/components/nodes/task/index.js:129]: Deferred as it's low return but fairly involved, so leaving it not DRY

**Review by vehikl-jacare — COMMENTED** [review]
@caleeli We've addressed the loss of reactivity that you found, thanks for testing it

**Comment by caleeli** [comment]
@vehikl-jacare there is something more to check, it fails after you change focus.
RenameTask mp4

**Comment by caleeli** [comment]
Id Validation is triggered when a Loop characteristic is added.
IdValidationLoop mp4

**Comment by vehikl-jacare** [comment]
@caleeli The ID validation will need to be updated in the [bpmnlint-plugin package](https://github.com/ProcessMaker/bpmnlint-plugin). We've created a ticket for it in that repo: https://github.com/ProcessMaker/bpmnlint-plugin/issues/7.

If there are no other issues this can be merged independent of that validation.

**Review by caleeli — APPROVED** [review]

**Pull request merged**